config/src/main/java/com/typesafe/config/ConfigDocumentFactory.java

```diff
@@ -0,0 +1,63 @@
+package com.typesafe.config;
+
+import com.typesafe.config.impl.ConfigImpl;
+import com.typesafe.config.impl.Parseable;
+
+import java.io.File;
+
+/**
+ * Factory for automatically creating a ConfigDocument from a given input. Currently
+ * only supports files and strings.
+ */
```

Comment:
we might want Reader too I suppose. Then people can plug in other kinds of stream.